### PR TITLE
Issue 6512: Standalone config properties should be included when create local segment store

### LINKED ISSUE #6512: standalone-config.properties should be included when standalone create the segmentstore
**Describe the bug**
In https://github.com/pravega/pravega/issues/6511, we know we can specify the `JAVA_OPTS` to configure the `publishedIPAddress`,  the problem is that passing configuration to system property is the only option we can choose to configure the segmentstore. Not all of configuration in `standalone-config.properties` could take effect as the standalone only include the configuration from the system property:
https://github.com/pravega/pravega/blob/a5aa4a29e5dbebc7fec6c080d4aca1c23ac9c5e4/standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java#L293

I know there is a workaround for that, but the configuration in `standalone-config.properties` should take effect, otherwise it will be very stranger. 

**How to fix**
I think adding the following code will work.
```
                .include(new Properties("./config/standalone-config.properties") )
```

### PR DESCRIPTION
**Pull request by CharlieLChen**
**Change log description**  
1. `standalone.config.properties` will be included when creating the local segment store
2. configuration from system properties will take higher priority to take effect.
3. Auto-formatting (idea CTRL+ALT+L)

**Purpose of the change**  
Fixes #6512

**What the code does**  
including `standalone.config.properties` when creating the local segment store, so that the configuration from that file could take effect.

**How to verify it**  

**Scenario 1:**
Add `pravegaservice.service.published.host.nameOrIp=10.247.101.103` to `standalone.config.properties`, then start the standalone, you check the log, you can see the `publishedIP` has changed, and you can write/read data from the remote client.
```shell
2022-01-11 15:28:56,588 3514 [main] INFO  i.p.s.server.host.ServiceStarter - ServiceConfig(containerCount: 1, publishedIPAddress: 10.247.101.103
```

**Scenario 2:**
Add `pravegaservice.service.published.host.nameOrIp=10.247.101.104` to `standalone.config.properties`, then export `JAVA_OPTS` like the below:
```
 export JAVA_OPTS="-Dpravegaservice.service.published.host.nameOrIp=10.247.101.103"
```
You will see the configuration from system properties overwrite the configuration from  `standalone.config.properties`

### CODE REVIEW

**Comment by fyang86** [comment]
hi @CharlieLChen , is it much more concise to just remove this line https://github.com/pravega/pravega/blob/f5ccb4c62087d37fe94fd116aa3c1e50b79d71b6/standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java#L310 to make it work?
With no `LISTENING_IP_ADDRESS` being configured, it will call `getHostAddress()` and assign to both `listeningIPAddress` and `publishedIPAddress` so that you don't need to change `standalone.config.properties` or set env variable everytime when starting a standalone cluster

**Comment by CharlieLChen** [comment]
> getHostAddress

`getHostAddress()` in UNIX-like system may return `127.0.0.1`. Besides that, I think `.with(ServiceConfig.LISTENING_IP_ADDRESS, ALL_INTERFACES)` is necessary, it means that: "I have multiple NIC, you can choose any of them to access me"

This is not only for remote access issues but for the best practice that config file should take effect, fixing the access issue is just a bonus.

**Review by tkaitchuck — APPROVED** [review]

**Comment by CharlieLChen** [comment]
It seems like we have trouble with UT. The failure seems to be not related to my change. I find even the master branch can't pass the Other UT and IT.

**Comment by CharlieLChen** [comment]
@tkaitchuck Can you please merge this PR if you think it's good? The Chinese Spring Festival is coming soon, I want this PR to be merged before I take a vacation. Thanks a lot.

**Review by tkaitchuck — APPROVED** [review]

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright Pravega Authors.
- *
+ * <p>
```
Comment (RaulGracia) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:3]: Do not change the license.

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 2]
```diff
@@ -47,6 +47,7 @@
 import io.pravega.segmentstore.storage.impl.bookkeeper.ZooKeeperServiceRunner;
 import io.pravega.shared.metrics.MetricsConfig;
 import io.pravega.shared.security.auth.Credentials;
+
```
Comment (RaulGracia) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:50]: Unnecessary change, please remove.

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 3]
```diff
@@ -66,6 +67,7 @@
 import org.apache.curator.framework.CuratorFrameworkFactory;
 import org.apache.curator.retry.ExponentialBackoffRetry;
 
+
```
Comment (RaulGracia) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:70]: Same here.

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 4]
```diff
@@ -167,10 +169,10 @@ public InProcPravegaCluster build() {
             //Check TLS related parameters
             Preconditions.checkState(!enableTls ||
                             (!Strings.isNullOrEmpty(this.keyFile)
-                            && !Strings.isNullOrEmpty(this.certFile)
-                            && !Strings.isNullOrEmpty(this.jksKeyFile)
-                            && !Strings.isNullOrEmpty(this.jksTrustFile)
-                            && !Strings.isNullOrEmpty(this.keyPasswordFile)),
+                                    && !Strings.isNullOrEmpty(this.certFile)
```
Comment (RaulGracia) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:172]: Same here, this change is not needed.

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java [hunk 5]
```diff
@@ -328,18 +330,24 @@ private void startLocalSegmentStore(int segmentStoreId) throws Exception {
...
+                        .with(AutoScalerConfig.TLS_CERT_FILE, this.certFile)
+                        .with(AutoScalerConfig.VALIDATE_HOSTNAME, false))
                 .include(MetricsConfig.builder()
                         .with(MetricsConfig.ENABLE_STATISTICS, enableMetrics)
                         .with(MetricsConfig.ENABLE_INFLUXDB_REPORTER, enableInfluxDB)
                         .with(MetricsConfig.ENABLE_PROMETHEUS, enablePrometheus)
                         .with(MetricsConfig.OUTPUT_FREQUENCY, metricsReportInterval));
 
+        String configFile = System.getProperty(SingleNodeConfig.PROPERTY_FILE);
+        if (configFile != null) {
```
Comment (RaulGracia) [inline standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:346]: Can you add a test for this?

**Review by RaulGracia — CHANGES_REQUESTED** [review]

**Comment by CharlieLChen** [comment]
Hi @RaulGracia, I have 2 questions about your comments: 
1. Some changes you think are unnecessary are made by Intellij auto-formatting(CTRL+ALT+L), just want to confirm is there enough reason to leave these unformatting codes alone? 
2. For now, there is no unit test for standalone, adding unit tests from 0 will be a huge task, that will be beyond the capacity of one PR,  should we create an issue to build unit tests for standalone from 0? After that, we can make a rule that adds a unit test for every change. Can we merge this PR firstly, then add unit test later?  As for the code quality, I think my change is quite straightforward and I have already tested it on alive standalone.

File: standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java (hunk 2 above)
Comment (CharlieLChen) [reply standalone/src/main/java/io/pravega/local/InProcPravegaCluster.java:50]: it's added by IntelliJ, I think it's necessary, you can tell this extra blank line separates the imports into 2 parts:
the first part is about the Pravega, the second part is about the java library.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**